ui/src/api/api.ts

```diff
@@ -103,6 +103,85 @@ export async function getVersion() {
   return result.build.version;
 }
 
+interface Tree {
+  [key: string]: any;
+}
+/**
+ * Fetch all metric values on one go using the list.
+ */
+export async function getMetricsAll() {
+  try {
+    const metrics = await getMetrics();
+    const paths = Object.keys(metrics);
+    // FIXME: path.forEach is more appropiate
+    await Promise.all(
+      paths.map(async (path) => {
+        const response = await getMetric(path);
+        metrics[path] = response;
+        return { path: response };
+      })
+    );
+
+    return metrics;
+  } catch (error) {
+    console.error("Error:", error);
+  }
+  return {};
+}
+
+/**
+ * Get list of metrics.
+ */
+async function getMetrics() {
+  const path = "/actuator/metrics";
+  return await get(path)
+    .then((data) => {
+      // Check if the data has 'names' property
+      if (data.hasOwnProperty("names")) {
```

Comment:
Why do we do we alter the output here and don't we create the right output in the backend? (just curious, you probably have a good reason)